src/DynamoCore/Models/PresetsState.cs

```diff
@@ -0,0 +1,109 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Dynamo.Core;
+using Dynamo.Services;
+using System.Xml;
+using Dynamo.Nodes;
+
+namespace Dynamo.Models
+{
+    /// <summary>
+    /// a class that saves the state of a subset of a graph
+    /// </summary>
+    public class PresetState
+    {
+        private Guid guid;
+        private readonly List<NodeModel> nodes;
+        private readonly List<XmlElement> serializedNodes;
+
+        # region properties
+
+        public string Name { get; private set; }
+        public string Description { get; private set; }
+       
+        /// <summary>
+        /// list of nodemodels that this state serializes
+        /// </summary>
+        public IEnumerable<NodeModel> Nodes { get{return nodes;}}
+
+        /// <summary>
+        /// list of serialized nodes
+        /// </summary>
+        public IEnumerable<XmlElement> SerializedNodes { get { return serializedNodes; }}
+
+        /// <summary>
+        /// A unique identifier for the state.
+        /// </summary>
+        public Guid Guid
+        {
+            get { return guid; }
+        }
+
+        #endregion
+
+        #region constructor
+        public PresetState(string name, string description, IEnumerable<NodeModel> inputsToSave, Guid id)
+        {
+            //if we have not supplied a guid at construction then create a new one
+            if (id == Guid.Empty)
+            {
+                guid = Guid.NewGuid();
+            }
+            else
+            {
+                guid = id;
+            }
+            
+            if (String.IsNullOrEmpty(name))
+            {
+                throw new ArgumentNullException("preset options state name is null");
+            }
+
+            if (inputsToSave == null || inputsToSave.Count() < 1)
+            {
+                throw new ArgumentNullException("nodes to save are null null");
+            } 
```

Comment:
`null null`?  Is that more `null` than `null`?  :)  Typically the arguments to `ArgumentNullException` are the name of the argument.  Further, you should put these checks right at the top of this method.